crypto/x509v3/v3_purp.c

```diff
@@ -871,6 +871,20 @@ const ASN1_OCTET_STRING *X509_get0_authority_key_id(X509 *x)
     return (x->akid != NULL ? x->akid->keyid : NULL);
 }
 
+const GENERAL_NAMES *X509_get0_authority_issuer(X509 *x)
+{
+    /* Call for side-effect of computing hash and caching extensions */
+    X509_check_purpose(x, -1, -1);
```

Comment:
every time I see this X509_check_purpose having the return value ignored, my eyes bleed.

Reply:
There is nothing much to check when `X509_check_purpose()` is called with -1 as argument.

https://github.com/openssl/openssl/blob/89a0ea064908eebb9048588864e1cc8f70717c4d/crypto/x509v3/v3_purp.c#L78-L87

IMHO it would have been better to call `x509v3_cache_extensions(x)` explicitly instead of doing such a hack which requires adding an explaining comment like in line 876 on every use.

Reply:
Looks like a [code health] task.

Reply:
Yes, and there are many things that can go wrong in x509v3_cache_extensions,
including memory allocations.
So that would deserve a return code.